app/src/integrationTest/kotlin/maru/app/MaruFollowerTest.kt

```diff
@@ -236,15 +237,140 @@ class MaruFollowerTest {
     checkValidatorAndFollowerBlocks(blocksToProduce * 2)
   }
 
+  @Test
+  fun `Maru follower is able to complete initial syncing`() {
+    followerStack.maruApp.stop()
+    followerStack.maruApp.close()
+    val blocksToProduce = 20
+    repeat(blocksToProduce) {
+      transactionsHelper.run {
+        validatorStack.besuNode.sendTransactionAndAssertExecution(
+          logger = log,
+          recipient = createAccount("another account"),
+          amount = Amount.ether(100),
+        )
+      }
+    }
+
+    // This is here mainly to wait until block propagation is complete
+    checkNetworkStackBlocksProduced(validatorStack, blocksToProduce)
+
+    followerStack.setMaruApp(
+      maruFactory.buildTestMaruFollowerWithP2pPeering(
+        ethereumJsonRpcUrl = followerStack.besuNode.jsonRpcBaseUrl().get(),
+        engineApiRpc = followerStack.besuNode.engineRpcUrl().get(),
+        dataDir = followerStack.tmpDir,
+        validatorPortForStaticPeering = validatorStack.p2pPort,
+      ),
+    )
+    followerStack.maruApp.start()
+
+    checkValidatorAndFollowerBlocks(blocksToProduce)
+  }
+
+  @Test
+  fun `Maru follower is able to complete syncing after restarted`() {
+    val blocksToProduce = 20
```

Comment:
Imo, setting `blocksToProduce = 10` should be enough for this test, as we will do two rounds